common/include/pcl/common/tuple_utils.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *
+ */
+
+#pragma once
+
+#include <pcl/type_traits.h>
+
+#include <type_traits>
+#include <tuple>
+
+namespace pcl {
+
+// for_each_tuple_until
+// Iterate over tuple
+// https://stackoverflow.com/questions/26902633/how-to-iterate-over-a-stdtuple-in-c-11
+template <typename TupleType, typename FunctionType>
+void
+for_each_until_true(
+    TupleType&&,
+    FunctionType,
+    std::integral_constant<
+        std::size_t,
+        std::tuple_size<typename std::remove_reference<TupleType>::type>::value>)
+{}
+
+template <std::size_t I,
+          typename TupleType,
+          typename FunctionType,
+          typename = typename std::enable_if<
+              I != std::tuple_size<
+                       typename std::remove_reference<TupleType>::type>::value>::type>
+void
+for_each_until_true(TupleType&& t, FunctionType f, std::integral_constant<size_t, I>)
+{
+  bool exit = f(std::get<I>(std::forward<TupleType>(t)));
+
+  if (!exit)
+    for_each_until_true(
+        std::forward<TupleType>(t), f, std::integral_constant<size_t, I + 1>());
+}
+
+template <typename TupleType, typename FunctionType>
+void
+for_each_until_true(TupleType&& t, FunctionType f)
+{
+  for_each_until_true(
+      std::forward<TupleType>(t), f, std::integral_constant<size_t, 0>());
+}
+
+// tuple_contains_type
+namespace detail {
+
+template <typename T, typename Tuple>
+struct tuple_contains_type_impl;
```

Comment:
Is this declaration for exposition? If so please mark as such

Reply:
> Is this declaration for exposition?
Didn't follow what you meant.